.circleci/config.yml

```diff
@@ -1,50 +1,213 @@
-# Java Gradle CircleCI 2.0 configuration file
-#
-# See https://github.com/grails-guides/grails-on-circleci-basics/blob/master/complete/.circleci/config.yml
+version: 2.1
 
-version: 2
-jobs:
-  gradle_test:
-    working_directory: ~/repo
+executors:
+  cloud-sdk:
     docker:
-    - image: circleci/openjdk:8-jdk
-    environment:
-      JVM_OPTS: -Xmx3200m
-      TERM: dumb
+      - image: google/cloud-sdk
+commands:
+  build-env:
+    description: "Build/push image to GCR"
+    parameters:
+      sa_key_name:
+        type: string
+        default: "GCLOUD_ORSP_DEV"
+      env:
+        type: string
+        default: "dev"
+      repo_name:
+        type: string
+        default: ${CIRCLE_PROJECT_REPONAME}
+      repo_branch:
+        type: string
+        default: ${CIRCLE_BRANCH}
+      google_project:
+        type: string
+        default: "orsp-dev"
+    steps:
+      - run:
+          name: Export env vars for substitution. $BASH_ENV is sourced before each step
+          command: |
+            echo 'export COMMIT="`git rev-parse --short HEAD`"' >> $BASH_ENV
+            echo 'export ENV="<< parameters.env >>"' >> $BASH_ENV
+            echo 'export REPO_NAME="<< parameters.repo_name >>"' >> $BASH_ENV
+            echo 'export REPO_BRANCH="<< parameters.repo_branch >>"' >> $BASH_ENV
```

Comment:
Do we actually use this anywhere?

Reply:
not yet, was intended to be used in env selection. Will remove if not used in a next PR.